Create README.md

NataljaMamonenko opened the pull request.

NataljaMamonenko committed:
Create README.md

NataljaMamonenko committed:
Add files via upload

NataljaMamonenko committed:
Add files via upload

irinakolody commented on PersonTest.java:
```diff
@@ -0,0 +1,38 @@
+    public void testGetFirstName(Person person, String expectedFirstName) {
+        String firstName = person.getFirstName();
+        Assert.assertEquals(firstName, expectedFirstName);
+    }
```
у вас тут буде помилка при запуску, бо ваш датапровайдер повертає 4 параметри:  return new Object[][]{{person, "John", "Doe", 30}};
а ви використовуєте тільки 2 (peson  та expectedFirstName)

аналогічно усюди, де ви використовуєте personProvider.


Взагалі, на логіку датапровайдеру найбільше лягає перевірка isRetired, де ви вкажете декілька жінок/ чоловіків різного віку і будете повертати очікуваний результат:


```
  return new Object[][] {
                { new Woman("Ann", "Duck", 23), false},
                { new Woman("Mary", "John", 60),  true}, 
                { new Woman("Mary", "John", 65), true}
        };
```

NataljaMamonenko committed:
2 attempt

NataljaMamonenko committed:
3 attempt

NataljaMamonenko committed:
Delete pom.xml

NataljaMamonenko committed:
Delete ManTest.java

NataljaMamonenko committed:
4 attempt

NataljaMamonenko committed:
Delete WomanTest.java

NataljaMamonenko committed:
Delete PersonTest.java

NataljaMamonenko committed:
5 attempt
rewrote tests ManTest, WomanTest, PersonTest
added @DataProviders methods in own classes
added GettersTest.xml

irinakolody commented on src/test/resources/GettersTest.xml:
```diff
@@ -0,0 +1,13 @@
+        <class name="PersonTest"/>
+
+    </classes>
+</test>
```
Структура файлу повинна бути збережена
```
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >

<suite name="Getters tests">
    <groups>
        <run>
            <include name="GettersTest"/>
        </run>
    </groups>
    <test name="Setters and Getters">
        <classes>
            <class name="WomanTest"/>
            <class name="ManTest"/>
            <class name="PersonTest"/>
        </classes>
    </test>
</suite>

```

irinakolody commented on src/test/java/ManTest.java:
```diff
@@ -1,67 +1,57 @@
         Woman partner = man.getPartner();
-        Assert.assertEquals(partner, woman);
+
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(partner, woman, "Partner is not retrieved correctly");
+        softAssert.assertAll();
     }
```
якщо один ассерт у тесті можна залишити просто  Assert.assertEquals(partner, woman, "Partner is not retrieved correctly");

irinakolody commented on src/test/java/WomanTest.java:
```diff
@@ -1,85 +1,81 @@
-        softAssert.assertTrue(woman.isRetired());
+        softAssert.assertEquals(partner, man);
         softAssert.assertAll();
     }
```
можна зробити аналогічно до testIsRetired в  ManTest: через датапровайдер.
